Framework Error

In MicroProfile Starter GH Actions and Circle CI build, the following error surfaced regularly. See [here](https://github.com/eclipse/microprofile-starter/actions/runs/415183661) for logs.
```
Extracting files to /home/runner/wlpExtract/liberty_2294323321101/wlp
Successfully extracted all product files.
Launching liberty (Open Liberty 20.0.0.12/wlp-1.0.47.cl201220201111-0736) on OpenJDK 64-Bit Server VM, version 11.0.9.1+1 (en_US)
[AUDIT   ] CWWKE0001I: The server liberty has been launched.
[AUDIT   ] CWWKZ0058I: Monitoring dropins for applications.
[AUDIT   ] CWWKS4104A: LTPA keys created in 1.519 seconds. LTPA key file: /home/runner/wlpExtract/liberty_2294323321101/wlp/usr/servers/liberty/resources/security/ltpa.keys
[AUDIT   ] CWPKI0803A: SSL certificate created in 1.458 seconds. SSL key file: /home/runner/wlpExtract/liberty_2294323321101/wlp/usr/servers/liberty/resources/security/public.jks
[AUDIT   ] CWWKT0016I: Web application available (default_host): http://localhost:9080/metrics/
[AUDIT   ] CWWKT0016I: Web application available (default_host): http://localhost:9080/jwt/
[AUDIT   ] CWWKT0016I: Web application available (default_host): http://localhost:9080/openapi/ui/
[AUDIT   ] CWWKT0016I: Web application available (default_host): http://localhost:9080/health/
[AUDIT   ] CWWKT0016I: Web application available (default_host): http://localhost:9080/openapi/
[AUDIT   ] CWWKT0016I: Web application available (default_host): http://localhost:9080/ibm/api/
[AUDIT   ] CWWKT0016I: Web application available (default_host): http://localhost:9080/
[AUDIT   ] CWWKZ0001I: Application liberty started in 3.020 seconds.
[AUDIT   ] CWWKF0012I: The server installed the following features: [appSecurity-2.0, cdi-2.0, concurrent-1.0, distributedMap-1.0, jaxrs-2.1, jaxrsClient-2.1, jndi-1.0, json-1.0, jsonb-1.0, jsonp-1.1, jwt-1.0, microProfile-3.3, monitor-1.0, mpConfig-1.4, mpFaultTolerance-2.1, mpHealth-2.2, mpJwt-1.1, mpMetrics-2.3, mpOpenAPI-1.1, mpOpenTracing-1.3, mpRestClient-1.4, opentracing-1.3, servlet-4.0, ssl-1.0].
[AUDIT   ] CWWKF0011I: The liberty server is ready to run a smarter planet. The liberty server started in 10.393 seconds.
[AUDIT   ] CWWKE0085I: The server liberty is stopping because the JVM is exiting.
[err] !SESSION 2020-12-11 10:34:50.100 -----------------------------------------------
eclipse.buildId=unknown
java.version=11.0.9.1
java.vendor=Oracle Corporation
BootLoader constants: OS=linux, ARCH=x86_64, WS=gtk, NL=en_US
!ENTRY org.eclipse.osgi 4 0 2020-12-11 10:34:50.101
!MESSAGE FrameworkEvent ERROR
!STACK 0
java.io.IOException: Exception in opening zip file: /home/runner/wlpExtract/liberty_2294323321101/wlp/lib/org.eclipse.osgi_3.15.0.jar
	at org.eclipse.osgi.framework.util.SecureAction.getZipFile(SecureAction.java:340)
	at org.eclipse.osgi.storage.bundlefile.ZipBundleFile.doOpen(ZipBundleFile.java:48)
	at org.eclipse.osgi.storage.bundlefile.CloseableBundleFile.open(CloseableBundleFile.java:128)
	at org.eclipse.osgi.storage.bundlefile.CloseableBundleFile.lockOpen(CloseableBundleFile.java:77)
	at org.eclipse.osgi.storage.bundlefile.CloseableBundleFile.getEntry(CloseableBundleFile.java:261)
	at org.eclipse.osgi.storage.BundleInfo$Generation.getEntry(BundleInfo.java:397)
	at org.eclipse.osgi.internal.framework.EquinoxBundle.getEntry(EquinoxBundle.java:701)
	at com.ibm.ws.artifact.url.internal.NotABundleResourceURLConnection.getManifestKeyForBundle(NotABundleResourceURLConnection.java:48)
	at com.ibm.ws.artifact.url.internal.NotABundleResourceURLConnection.forgetBundle(NotABundleResourceURLConnection.java:96)
	at com.ibm.ws.artifact.url.internal.URLStreamHandlerServiceImpl.addingBundle(URLStreamHandlerServiceImpl.java:56)
	at com.ibm.ws.artifact.url.internal.URLStreamHandlerServiceImpl.addingBundle(URLStreamHandlerServiceImpl.java:29)
	at org.osgi.util.tracker.BundleTracker$Tracked.customizerAdding(BundleTracker.java:475)
	at org.osgi.util.tracker.BundleTracker$Tracked.customizerAdding(BundleTracker.java:1)
	at org.osgi.util.tracker.AbstractTracked.trackAdding(AbstractTracked.java:256)
	at org.osgi.util.tracker.AbstractTracked.track(AbstractTracked.java:229)
	at org.osgi.util.tracker.BundleTracker$Tracked.bundleChanged(BundleTracker.java:450)
	at org.eclipse.osgi.internal.framework.BundleContextImpl.dispatchEvent(BundleContextImpl.java:973)
	at org.eclipse.osgi.framework.eventmgr.EventManager.dispatchEvent(EventManager.java:234)
	at org.eclipse.osgi.framework.eventmgr.ListenerQueue.dispatchEventSynchronous(ListenerQueue.java:151)
	at org.eclipse.osgi.internal.framework.EquinoxEventPublisher.publishBundleEventPrivileged(EquinoxEventPublisher.java:235)
	at org.eclipse.osgi.internal.framework.EquinoxEventPublisher.publishBundleEvent(EquinoxEventPublisher.java:141)
	at org.eclipse.osgi.internal.framework.EquinoxEventPublisher.publishBundleEvent(EquinoxEventPublisher.java:133)
	at org.eclipse.osgi.internal.framework.EquinoxContainerAdaptor.publishModuleEvent(EquinoxContainerAdaptor.java:217)
	at org.eclipse.osgi.container.Module.publishEvent(Module.java:498)
	at org.eclipse.osgi.container.Module.doStop(Module.java:656)
	at org.eclipse.osgi.container.Module.stop(Module.java:520)
	at org.eclipse.osgi.container.SystemModule.stop(SystemModule.java:207)
	at org.eclipse.osgi.internal.framework.EquinoxBundle$SystemBundle$EquinoxSystemModule$1.run(EquinoxBundle.java:220)
	at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: java.nio.file.NoSuchFileException: /home/runner/wlpExtract/liberty_2294323321101/wlp/lib/org.eclipse.osgi_3.15.0.jar
	at java.base/sun.nio.fs.UnixException.translateToIOException(UnixException.java:92)
	at java.base/sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:111)
	at java.base/sun.nio.fs.UnixException.rethrowAsIOException(UnixException.java:116)
	at java.base/sun.nio.fs.UnixFileAttributeViews$Basic.readAttributes(UnixFileAttributeViews.java:55)
	at java.base/sun.nio.fs.UnixFileSystemProvider.readAttributes(UnixFileSystemProvider.java:149)
	at java.base/sun.nio.fs.LinuxFileSystemProvider.readAttributes(LinuxFileSystemProvider.java:99)
	at java.base/java.nio.file.Files.readAttributes(Files.java:1763)
	at java.base/java.util.zip.ZipFile$Source.get(ZipFile.java:1225)
	at java.base/java.util.zip.ZipFile$CleanableResource.<init>(ZipFile.java:727)
	at java.base/java.util.zip.ZipFile$CleanableResource.get(ZipFile.java:844)
	at java.base/java.util.zip.ZipFile.<init>(ZipFile.java:247)
	at java.base/java.util.zip.ZipFile.<init>(ZipFile.java:177)
	at java.base/java.util.zip.ZipFile.<init>(ZipFile.java:191)
	at org.eclipse.osgi.framework.util.SecureAction.getZipFile(SecureAction.java:322)
	... 28 more
[AUDIT   ] quiesce.begin
[AUDIT   ] CWWKT0017I: Web application removed (default_host): http://localhost:9080/
[AUDIT   ] CWWKZ0009I: The application liberty has stopped successfully.
[AUDIT   ] CWWKT0017I: Web application removed (default_host): http://localhost:9080/metrics/
[AUDIT   ] CWWKT0017I: Web application removed (default_host): http://localhost:9080/health/
[AUDIT   ] CWWKT0017I: Web application removed (default_host): http://localhost:9080/openapi/
[AUDIT   ] CWWKT0017I: Web application removed (default_host): http://localhost:9080/openapi/ui/
[AUDIT   ] CWWKT0017I: Web application removed (default_host): http://localhost:9080/jwt/
[AUDIT   ] CWWKT0017I: Web application removed (default_host): http://localhost:9080/ibm/api/
[AUDIT   ] CWWKE0036I: The server liberty stopped after 12.773 seconds.


```

I also found some other exceptions, which might be related to this.
`java.lang.AssertionError: Runtime log service-b-run.log log should not contain error or warning lines that are not whitelisted. See target/archived-logs/org.eclipse.microprofile.starter.TestMatrixTest/libertyJWTRest/service-b-run.log and check these offending lines: 
[ERROR   ] CWWKE0701E: bundle com.ibm.ws.dynacache:1.0.47.cl201220201111-0736 (69)[com.ibm.ws.cache.CacheServiceImpl(202)] : The stop method has thrown an exception java.lang.NoClassDefFoundError: com/ibm/ws/cache/NonSyncHashtableEnumerator
[ERROR   ] CWWKE0701E: bundle com.ibm.ws.channelfw:1.0.47.cl201220201111-0736 (91)[com.ibm.ws.channelfw.internal.CHFWEventHandler(206)] : Could not load implementation object class com.ibm.ws.channelfw.internal.CHFWEventHandler java.lang.ClassNotFoundException: com.ibm.ws.channelfw.internal.CHFWEventHandler cannot be found by com.ibm.ws.channelfw_1.0.47.cl201220201111-0736
[ERROR   ] CWWKE0701E: bundle com.ibm.ws.runtime.update:1.0.47.cl201220201111-0736 (16)[com.ibm.ws.runtime.update.internal.PauseableComponentQuiesceListener(15)] : Could not load implementation object class com.ibm.ws.runtime.update.internal.PauseableComponentQuiesceListener java.lang.ClassNotFoundException: com.ibm.ws.runtime.update.internal.PauseableComponentQuiesceListener cannot be found by com.ibm.ws.runtime.update_1.0.47.cl201220201111-0736
[ERROR   ] CWWKE0701E: bundle com.ibm.ws.channelfw:1.0.47.cl201220201111-0736 (91)[com.ibm.websphere.channelfw.osgi.CHFWBundle(207)] : The unsetFactoryProvider method has thrown an exception java.lang.NoClassDefFoundError: com/ibm/ws/channelfw/internal/ChannelFrameworkImpl$1
[ERROR   ] CWWKE0701E: bundle com.ibm.ws.channelfw:1.0.47.cl201220201111-0736 (91)[com.ibm.websphere.channelfw.osgi.CHFWBundle(207)] : The deactivate method has thrown an exception java.lang.NoClassDefFoundError: com/ibm/ws/channelfw/internal/ChannelFrameworkImpl$1
[ERROR   ] CWWKE0701E: bundle com.ibm.ws.channel.ssl:1.0.47.cl201220201111-0736 (95)[com.ibm.ws.channel.ssl.internal.SSLChannelProvider(413)] : The deactivate method has thrown an exception java.lang.NoClassDefFoundError: com/ibm/ws/channelfw/internal/ChannelFrameworkImpl$1
	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.assertTrue(Assert.java:41)
	at org.eclipse.microprofile.starter.utils.Logs.checkLog(Logs.java:87)
	at org.eclipse.microprofile.starter.TestMatrixTest.testRuntime(TestMatrixTest.java:178)
	at org.eclipse.microprofile.starter.TestMatrixTest.libertyJWTRest(TestMatrixTest.java:398)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.jboss.arquillian.junit.Arquillian$8$1.invokeMethod(Arquillian.java:325)
	at org.jboss.arquillian.junit.MethodInvoker$1.invoke(MethodInvoker.java:18)
	at org.jboss.arquillian.container.test.impl.execution.LocalTestExecuter.execute(LocalTestExecuter.java:57)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.jboss.arquillian.core.impl.ObserverImpl.invoke(ObserverImpl.java:86)
	at org.jboss.arquillian.core.impl.EventContextImpl.invokeObservers(EventContextImpl.java:103)
	at org.jboss.arquillian.core.impl.EventContextImpl.proceed(EventContextImpl.java:90)
	at org.jboss.arquillian.core.impl.ManagerImpl.fire(ManagerImpl.java:133)
	at org.jboss.arquillian.core.impl.ManagerImpl.fire(ManagerImpl.java:105)
	at org.jboss.arquillian.core.impl.EventImpl.fire(EventImpl.java:62)
	at org.jboss.arquillian.container.test.impl.execution.ClientTestExecuter.execute(ClientTestExecuter.java:50)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.jboss.arquillian.core.impl.ObserverImpl.invoke(ObserverImpl.java:86)
	at org.jboss.arquillian.core.impl.EventContextImpl.invokeObservers(EventContextImpl.java:103)
	at org.jboss.arquillian.core.impl.EventContextImpl.proceed(EventContextImpl.java:90)
	at org.jboss.arquillian.container.test.impl.client.ContainerEventController.createContext(ContainerEventController.java:128)
	at org.jboss.arquillian.container.test.impl.client.ContainerEventController.createTestContext(ContainerEventController.java:118)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.jboss.arquillian.core.impl.ObserverImpl.invoke(ObserverImpl.java:86)
	at org.jboss.arquillian.core.impl.EventContextImpl.proceed(EventContextImpl.java:95)
	at org.jboss.arquillian.test.impl.TestContextHandler.createTestContext(TestContextHandler.java:116)
	at jdk.internal.reflect.GeneratedMethodAccessor19.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.jboss.arquillian.core.impl.ObserverImpl.invoke(ObserverImpl.java:86)
	at org.jboss.arquillian.core.impl.EventContextImpl.proceed(EventContextImpl.java:95)
	at org.jboss.arquillian.test.impl.TestContextHandler.createClassContext(TestContextHandler.java:83)
	at jdk.internal.reflect.GeneratedMethodAccessor18.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.jboss.arquillian.core.impl.ObserverImpl.invoke(ObserverImpl.java:86)
	at org.jboss.arquillian.core.impl.EventContextImpl.proceed(EventContextImpl.java:95)
	at org.jboss.arquillian.test.impl.TestContextHandler.createSuiteContext(TestContextHandler.java:69)
	at jdk.internal.reflect.GeneratedMethodAccessor17.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.jboss.arquillian.core.impl.ObserverImpl.invoke(ObserverImpl.java:86)
	at org.jboss.arquillian.core.impl.EventContextImpl.proceed(EventContextImpl.java:95)
	at org.jboss.arquillian.core.impl.ManagerImpl.fire(ManagerImpl.java:133)
	at org.jboss.arquillian.test.impl.EventTestRunnerAdaptor.test(EventTestRunnerAdaptor.java:139)
	at org.jboss.arquillian.junit.MethodInvoker.invoke(MethodInvoker.java:15)
	at org.jboss.arquillian.junit.Arquillian$8.evaluate(Arquillian.java:332)
	at org.jboss.arquillian.junit.Arquillian$4.evaluate(Arquillian.java:204)
	at org.jboss.arquillian.junit.Arquillian.multiExecute(Arquillian.java:350)
	at org.jboss.arquillian.junit.Arquillian.access$200(Arquillian.java:54)
	at org.jboss.arquillian.junit.Arquillian$5.evaluate(Arquillian.java:215)
	at org.junit.rules.TestWatcher$1.evaluate(TestWatcher.java:55)
	at org.junit.rules.RunRules.evaluate(RunRules.java:20)
	at org.jboss.arquillian.junit.Arquillian$7$1.invoke(Arquillian.java:279)
	at org.jboss.arquillian.container.test.impl.execution.ClientBeforeAfterLifecycleEventExecuter.execute(ClientBeforeAfterLifecycleEventExecuter.java:88)
	at org.jboss.arquillian.container.test.impl.execution.ClientBeforeAfterLifecycleEventExecuter.on(ClientBeforeAfterLifecycleEventExecuter.java:66)
	at jdk.internal.reflect.GeneratedMethodAccessor23.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.jboss.arquillian.core.impl.ObserverImpl.invoke(ObserverImpl.java:86)
	at org.jboss.arquillian.core.impl.EventContextImpl.invokeObservers(EventContextImpl.java:103)
	at org.jboss.arquillian.core.impl.EventContextImpl.proceed(EventContextImpl.java:90)
	at org.jboss.arquillian.container.test.impl.client.ContainerEventController.createContext(ContainerEventController.java:128)
	at org.jboss.arquillian.container.test.impl.client.ContainerEventController.createBeforeContext(ContainerEventController.java:114)
	at jdk.internal.reflect.GeneratedMethodAccessor22.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.jboss.arquillian.core.impl.ObserverImpl.invoke(ObserverImpl.java:86)
	at org.jboss.arquillian.core.impl.EventContextImpl.proceed(EventContextImpl.java:95)
	at org.jboss.arquillian.test.impl.TestContextHandler.createTestContext(TestContextHandler.java:116)
	at jdk.internal.reflect.GeneratedMethodAccessor19.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.jboss.arquillian.core.impl.ObserverImpl.invoke(ObserverImpl.java:86)
	at org.jboss.arquillian.core.impl.EventContextImpl.proceed(EventContextImpl.java:95)
	at org.jboss.arquillian.test.impl.TestContextHandler.createClassContext(TestContextHandler.java:83)
	at jdk.internal.reflect.GeneratedMethodAccessor18.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.jboss.arquillian.core.impl.ObserverImpl.invoke(ObserverImpl.java:86)
	at org.jboss.arquillian.core.impl.EventContextImpl.proceed(EventContextImpl.java:95)
	at org.jboss.arquillian.test.impl.TestContextHandler.createSuiteContext(TestContextHandler.java:69)
	at jdk.internal.reflect.GeneratedMethodAccessor17.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.jboss.arquillian.core.impl.ObserverImpl.invoke(ObserverImpl.java:86)
	at org.jboss.arquillian.core.impl.EventContextImpl.proceed(EventContextImpl.java:95)
	at org.jboss.arquillian.core.impl.ManagerImpl.fire(ManagerImpl.java:133)
	at org.jboss.arquillian.core.impl.ManagerImpl.fire(ManagerImpl.java:105)
	at org.jboss.arquillian.test.impl.EventTestRunnerAdaptor.fireCustomLifecycle(EventTestRunnerAdaptor.java:159)
	at org.jboss.arquillian.junit.Arquillian$7.evaluate(Arquillian.java:273)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.jboss.arquillian.junit.Arquillian$2.evaluate(Arquillian.java:166)
	at org.jboss.arquillian.junit.Arquillian.multiExecute(Arquillian.java:350)
	at org.jboss.arquillian.junit.Arquillian.access$200(Arquillian.java:54)
	at org.jboss.arquillian.junit.Arquillian$3.evaluate(Arquillian.java:177)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.jboss.arquillian.junit.Arquillian.run(Arquillian.java:115)
	at org.apache.maven.surefire.junit4.JUnit4Provider.execute(JUnit4Provider.java:252)
	at org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:141)
	at org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:112)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.apache.maven.surefire.util.ReflectionUtils.invokeMethodWithArray(ReflectionUtils.java:189)
	at org.apache.maven.surefire.booter.ProviderFactory$ProviderProxy.invoke(ProviderFactory.java:165)
	at org.apache.maven.surefire.booter.ProviderFactory.invokeProvider(ProviderFactory.java:85)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:115)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:75)`

--- a/dev/wlp.lib.extract/src/wlp/lib/extract/SelfExtractRun.java
+++ b/dev/wlp.lib.extract/src/wlp/lib/extract/SelfExtractRun.java
@@ -120,7 +120,7 @@ public class SelfExtractRun extends SelfExtract {
      * @return unique dir name
      */
     private static String createTempDirectory(String baseDir, String fileStem) {
-        Long nano = new Long(System.nanoTime());
+        Long nano = Long.valueOf(System.nanoTime());
         return baseDir + File.separator + fileStem + nano;
     }
 
@@ -271,7 +271,7 @@ public class SelfExtractRun extends SelfExtract {
         outputReader.start();
 
         // now setup the shutdown hook
-        Runtime.getRuntime().addShutdownHook(new Thread(new ShutdownHook(platformType, extractDirectory, serverName, outputReader, errorReader, extractDirPredefined)));
+        Runtime.getRuntime().addShutdownHook(new Thread(new ShutdownHook(platformType, extractDirectory, serverName, extractDirPredefined)));
         // wait on server start process to complete, capture and pass on return code
         rc = proc.waitFor();
 
@@ -349,7 +349,7 @@ public class SelfExtractRun extends SelfExtract {
         props.setProperty("LOG_DIR",
                           extractDirectory + File.separator + "wlp" + File.separator + "usr" + File.separator + "servers" + File.separator + serverName + File.separator + "logs");
 
-        Class clazz = cl.loadClass(className);
+        Class<?> clazz = cl.loadClass(className);
         List<String> argList = new ArrayList<String>(args.length + 2);
         argList.add(serverName);
         if (args.length > 0) {
@@ -359,7 +359,7 @@ public class SelfExtractRun extends SelfExtract {
         Method m = clazz.getDeclaredMethod("main", new Class[] { String[].class });
 
         // Add the shutdown hook to clean up
-        Runtime.getRuntime().addShutdownHook(new Thread(new ShutdownHook(platformType, extractDirectory, serverName, null, null, extractDirPredefined)));
+        Runtime.getRuntime().addShutdownHook(new Thread(new ShutdownHook(platformType, extractDirectory, serverName, extractDirPredefined)));
 
         attachJavaAgent(extractDirectory);
 
@@ -388,10 +388,11 @@ public class SelfExtractRun extends SelfExtract {
                 URL thisJar = SelfExtractRun.class.getProtectionDomain().getCodeSource().getLocation();
                 try {
                     URL toolsJar = new URL("file:" + f.getCanonicalPath());
+                    @SuppressWarnings("resource")
                     URLClassLoader cl = new URLClassLoader(new URL[] { thisJar, toolsJar }, null);
                     Class<?> clazz = cl.loadClass("wlp.lib.extract.AgentAttach");
                     Method m = clazz.getDeclaredMethod("attach", new Class[] { String.class });
-                    Object result = m.invoke(null, new String[] { javaAgent.getAbsolutePath() });
+                    Object result = m.invoke(null, javaAgent.getAbsolutePath());
                     if (result != null) {
                         err("UNABLE_TO_ATTACH_AGENT", result);
                     }

--- a/dev/wlp.lib.extract/src/wlp/lib/extract/ShutdownHook.java
+++ b/dev/wlp.lib.extract/src/wlp/lib/extract/ShutdownHook.java
@@ -30,39 +30,24 @@ public class ShutdownHook implements Runnable {
 
     private static final ResourceBundle resourceBundle = ResourceBundle.getBundle(SelfExtract.class.getName() + "Messages");
 
-    int platformType;
-    String dir;
-    String serverName;
-    Thread out;
-    Thread err;
-    boolean extractDirPredefined = false;
-
-    // no parm ctor not public
-    private ShutdownHook() {
-
-    }
+    final int platformType;
+    final String dir;
+    final String serverName;
+    final boolean extractDirPredefined;
 
     /**
      * The only constructor.
      *
-     * @param platformType - platform type: unix(1), windows(2), cygwin(3)
-     * @param dir - extraction directory
-     * @param serverName - name of server from jar (in extraction directory)
-     * @param out - output stream reader thread of parent process.
-     * @param err - error stream reader thread of parent process.
+     * @param platformType         - platform type: unix(1), windows(2), cygwin(3)
+     * @param dir                  - extraction directory
+     * @param serverName           - name of server from jar (in extraction directory)
      * @param extractDirPredefined - flag which indicates if WLP_JAR_EXTRACT_DIR was predefined by user
      */
     public ShutdownHook(int platformType,
                         String dir,
                         String serverName,
-                        StreamReader out,
-                        StreamReader err,
                         boolean extractDirPredefined) {
-
-        this();
         this.serverName = serverName;
-        this.out = out;
-        this.err = err;
         this.dir = dir;
         this.platformType = platformType;
         this.extractDirPredefined = extractDirPredefined;
@@ -155,7 +140,7 @@ public class ShutdownHook implements Runnable {
      * Write logic for windows cleanup script
      *
      * @param file - script File object
-     * @param bw - bufferedWriter to write into script file
+     * @param bw   - bufferedWriter to write into script file
      * @throws IOException
      */
     private void writeWindowsCleanup(File file, BufferedWriter bw) throws IOException {
@@ -191,7 +176,7 @@ public class ShutdownHook implements Runnable {
      * Write logic for Unix cleanup script
      *
      * @param file - script File object
-     * @param bw - bufferedWriter to write into script file
+     * @param bw   - bufferedWriter to write into script file
      * @throws IOException
      */
     private void writeUnixCleanup(File file, BufferedWriter bw) throws IOException {
@@ -224,7 +209,7 @@ public class ShutdownHook implements Runnable {
      * Write logic for Cygwin cleanup script
      *
      * @param file - script File object
-     * @param bw - bufferedWriter to write into script file
+     * @param bw   - bufferedWriter to write into script file
      * @throws IOException
      */
     private void writeCygwinCleanup(File file, BufferedWriter bw) throws IOException {
@@ -298,18 +283,6 @@ public class ShutdownHook implements Runnable {
         try {
 
             stopServer(); // first, stop server
-
-            // wait on error/output stream threads to complete
-            // note on Windows the streams never close, so wait with brief timeout
-            if (out != null && err != null) {
-                if (!System.getProperty("os.name").startsWith("Win")) {
-                    out.join();
-                    err.join();
-                } else { // windows, so use timeout
-                    out.join(500);
-                    err.join(500);
-                }
-            }
 
             // When the server is launched with java -jar, delete the server on exit minus
             // the /logs folder, unless WLP_JAR_EXTRACT_DIR is set at which point don't delete

--- a/dev/wlp.lib.extract/src/wlp/lib/extract/ShutdownHook.java
+++ b/dev/wlp.lib.extract/src/wlp/lib/extract/ShutdownHook.java
@@ -117,8 +117,12 @@ public class ShutdownHook implements Runnable {
             cmd = "bash -c  " + '"' + cmd.replace('\\', '/') + '"';
         }
 
-        Runtime.getRuntime().exec(cmd, SelfExtractUtils.runEnv(dir), null); // stop server
-
+        Process stopProcess = Runtime.getRuntime().exec(cmd, SelfExtractUtils.runEnv(dir), null); // stop server
+        try {
+            stopProcess.waitFor();
+        } catch (InterruptedException e) {
+            Thread.currentThread().interrupt();
+        }
     }
 
     /**

